frontend/fingertips-frontend/lib/search/indicatorMapper.test.ts

```diff
@@ -0,0 +1,243 @@
+            {
+              areaCode: 'Area1',
+              trend: HealthDataPointTrendEnum.Decreasing,
+            },
+            {
+              areaCode: 'Area2',
+              trend: HealthDataPointTrendEnum.NoSignificantChange,
+            },
+          ],
+          unitLabel: '',
+          hasInequalities: true,
+          usedInPoc: true,
+        },
+      ];
+
+      const result = indicatorMapper.toEntities(
+        mockRawIndicators,
+        ['Area1'],
+        true
+      );
+
+      expect(result).toEqual([
+        {
+          indicatorID: '1',
+          indicatorName: 'Red faced',
+          indicatorDefinition:
+            'Count of people who did something they are embarrassed by',
+          earliestDataPeriod: '1938',
+          latestDataPeriod: '2023',
+          dataSource: 'The Beano',
+          lastUpdatedDate: new Date('December 6, 2024'),
+          trend: 'Increasing and getting worse',
+          unitLabel: '',
+          hasInequalities: false,
+        },
+        {
+          indicatorID: '2',
+          indicatorName: 'Perp count',
+          indicatorDefinition: 'Perps brought to justice by Red Angel',
+          earliestDataPeriod: '1977',
+          latestDataPeriod: '2022',
+          dataSource: 'Mega City 1',
+          lastUpdatedDate: new Date('November 5, 2023'),
+          trend: 'Decreasing',
+          unitLabel: '',
+          hasInequalities: true,
+        },
+      ]);
+    });
+
+    it('maps trend to undefined when a user has requested more an indicator in more than one area', () => {
+      const mockRawIndicators = [
+        {
+          indicatorID: '1',
+          indicatorName: 'Red faced',
+          indicatorDefinition:
+            'Count of people who did something they are embarrassed by',
+          earliestDataPeriod: '1938',
+          latestDataPeriod: '2023',
+          dataSource: 'The Beano',
+          lastUpdatedDate: new Date('December 6, 2024'),
+          associatedAreaCodes: ['Area1', 'Area2'],
+          trendsByArea: [
+            {
+              areaCode: 'Area1',
+              trend: HealthDataPointTrendEnum.IncreasingAndGettingWorse,
+            },
+            {
+              areaCode: 'Area2',
+              trend: HealthDataPointTrendEnum.NoSignificantChange,
+            },
+          ],
+          unitLabel: '',
+          hasInequalities: false,
+          usedInPoc: true,
+        },
+      ];
+
+      const result = indicatorMapper.toEntities(
+        mockRawIndicators,
+        ['Area1', 'Area2'],
+        true
+      );
+
+      expect(result).toEqual([
+        {
+          indicatorID: '1',
+          indicatorName: 'Red faced',
+          indicatorDefinition:
+            'Count of people who did something they are embarrassed by',
+          earliestDataPeriod: '1938',
+          latestDataPeriod: '2023',
+          dataSource: 'The Beano',
+          lastUpdatedDate: new Date('December 6, 2024'),
+          trend: undefined,
```

Comment:
I'm a little confused by this. The test above for indicator 2 has more than one area but doesn't return undefined there. What makes that behave differently?

Reply:
ah I see it's line 121 where we are requesting 2 areas.